graphics/src/gradient.rs

```diff
@@ -0,0 +1,103 @@
+//! For creating a Gradient.
+use iced_native::Color;
+use crate::gradient::linear::Linear;
+use crate::Point;
+
+#[derive(Debug, Clone, PartialEq)]
+/// A fill which transitions colors progressively along a direction, either linearly, radially (TBD),
+/// or conically (TBD).
+pub enum Gradient {
+    /// A linear gradient interpolates colors along a direction from its [`start`] to its [`end`]
+    /// point.
+    Linear(Linear),
+}
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+/// A point along the gradient vector where the specified [`color`] is unmixed.
+pub struct ColorStop {
```

Comment:
Is just `Stop` still understandable in the `gradient` context? `gradient::Stop`...